ADR/ADR-001-Git-Branching-Model.md

```diff
@@ -0,0 +1,253 @@
+* Requires strong team discipline: Teams must ensure main is always deployable
+* Not as common in operational software: Many operational software packages use Git Flow
+
+### Analysis
+
+**Team Size Impact**
+
+* Small teams (1-3 developers): GitHub Flow reduces overhead and simplifies collaboration
+* Medium teams (4-8 developers): Either approach viable, depends on release cadence and software type
+* Large teams (9+ developers): Git Flow provides better coordination and parallel development structure
+
+**Release Patterns**
+
+* Continuous deployment: GitHub Flow aligns perfectly with continuous delivery practices
+* Scheduled releases: Git Flow provides better control and release preparation capabilities. Also supports synchronised releases of multiple components.
+* Mixed patterns: May require different strategies per repository based on deployment needs
+
+**Software Type Considerations**
+
+* Web applications: GitHub Flow ideal for continuously delivered web apps (when appropriate CI/CD is in place)
+* Versioned software: Git Flow better suited for software requiring multiple version support (especially if the software is operational)
+* Simple applications: GitHub Flow avoids unnecessary complexity
+
+**Operational Complexity**
+
+* Simple applications: GitHub Flow reduces unnecessary complexity
+* Critical systems: Git Flow provides additional safety measures and structured release process
+* Evolving requirements: Start simple, add complexity as needed
+
+## Decision
+
+We will adopt a **hybrid approach** that allows repositories to evolve their branching strategy based on operational needs and complexity:
+
+### Initial State: GitHub Flow
+All new repositories will start with GitHub Flow:
+* Use main/master as the primary branch
+* Create feature branches for development
+* Use pull requests for code review and collaboration
+* Merge directly to main when features are complete and reviewed
+* Deploy directly from a tag on main (manual or automated)
+* Use hotfix branches for critical production issues if needed
+
+### Evolution Trigger: GateKeeper Decision
+Lead developers (GateKeepers) for each repository can decide to upgrade to Git Flow when one or more of the following conditions are met:
+
+* Repository usage becomes operationally critical
+* Team size exceeds comfortable coordination limits (typically 8+ developers)
+* Software requires explicit versioning or multiple version support
+* Release process requires more structured approach with dedicated release preparation
+* Stakeholders require additional quality gates and formal release cycles
+* Deployment risk warrants additional safeguards and staging processes
+* Continuous delivery is not suitable for the software type
+
+The reverse can also happen, where a repository using Git Flow can revert to GitHub Flow if it the assessment of the criteria reverses (e.g. becomes simpler or less critical).
+
+### Migration Path
```

Comment:
how does a developer tell what approach is followed by a given repo? Should we create contributor-guideline template in the cookiecutter repo template? And link the two templates in here?

Reply:
I dont think that guarantees. As we see sometimes a branch exists and is not used. 
Branches are 'temporal' -- I would rely on proper documenting on each repository what is the branching model in use.